src/SingleArticle.jsx

```diff
@@ -16,6 +17,46 @@ export const SingleArticle = () => {
         })
     }, [])
 
+    const handleVote = (article_id, votes, increase) => {
+        setSingleArticle((currSingleArticle) => {
+            if(increase){
+                return {...currSingleArticle, votes: currSingleArticle.votes + 1}
+            }else{
+                return {...currSingleArticle, votes: currSingleArticle.votes - 1}
+            }
+            
+        })
+
+        updateArticleVotes(article_id, votes).catch((err) => {
+            setIsError(true)
+            setSingleArticle((currSingleArticle) => {
+                if(increase){
+                    return {...currSingleArticle, votes: currSingleArticle.votes - 1}
+                }else{
+                    return {...currSingleArticle, votes: currSingleArticle.votes + 1}
+                }
+            })
+        })
+        
+    }
+
+    if(isError){
```

Comment:
There is a lot of repeated code so you might want to refactor to make a bit more dry at some point.